pkg/scheduler/core/generic_scheduler.go

```diff
@@ -204,6 +228,11 @@ func (g *genericScheduler) Schedule(ctx context.Context, prof *profile.Profile,
 	host, err := g.selectHost(priorityList)
```

Comment:
should this `err` be recorded into the span?

Reply:
Good suggestion